payments-core/src/main/java/com/stripe/android/StripeIntentResult.kt

```diff
@@ -16,60 +19,75 @@ abstract class StripeIntentResult<out T : StripeIntent> internal constructor(
     abstract val intent: T
     abstract val failureMessage: String?
 
+    @VisibleForTesting
+    internal var luxeNextActionRepository: LuxeNextActionRepository = LuxeNextActionRepository.Instance
+
     @Outcome
     @get:Outcome
     val outcome: Int
-        get() = determineOutcome(intent.status, outcomeFromFlow)
+        get() = determineOutcome(intent, outcomeFromFlow)
 
     @StripeIntentResult.Outcome
     private fun determineOutcome(
-        stripeIntentStatus: StripeIntent.Status?,
+        stripeIntent: StripeIntent,
         @StripeIntentResult.Outcome outcome: Int
     ): Int {
         if (outcome != Outcome.UNKNOWN) {
             return outcome
         }
 
-        return when (stripeIntentStatus) {
-            StripeIntent.Status.RequiresAction -> {
-                if (isNextActionSuccessState(intent)) {
-                    Outcome.SUCCEEDED
-                } else {
+        return getOutcome(stripeIntent)
+    }
+
+    private fun getOutcome(stripeIntent: StripeIntent): Int {
+        return luxeNextActionRepository.getPostAuthorizeIntentOutcome(
+            stripeIntent
+        ) ?: run {
+            Log.e("MLB", "Doing the old way.")
+            when (stripeIntent.status) {
+                StripeIntent.Status.RequiresAction -> {
+                    if(stripeIntent.nextActionData == null){
```

Comment:
If we were not able to parse the next action and it is requires action, then we didn't do the next action step and this is a failure.